src/app/pages/data-protection/cloudsync/cloudsync-wizard/steps/cloudsync-provider/cloudsync-provider.component.ts

```diff
@@ -0,0 +1,277 @@
+      this.save.emit(this.existingCredential);
+      this.isLoading$.next(false);
+      this.cdr.markForCheck();
+      return false;
+    }
+
+    const beforeSubmit$ = this.providerForm.beforeSubmit();
+
+    beforeSubmit$
+      .pipe(
+        switchMap(() => {
+          const payload = this.getPayload();
+          return this.isNew
+            ? this.ws.call('cloudsync.credentials.create', [payload])
+            : this.ws.call('cloudsync.credentials.update', [this.existingCredential.id, payload]);
+        }),
+        untilDestroyed(this),
+      )
+      .subscribe({
+        next: (credential) => {
+          this.isLoading$.next(false);
+          this.snackbarService.success(
+            this.isNew
+              ? this.translate.instant('Cloud credential added.')
+              : this.translate.instant('Cloud credential updated.'),
+          );
+          this.save.emit(credential);
+          this.cdr.markForCheck();
+        },
+        error: (error) => {
+        // TODO: Errors for nested provider form will be shown in a modal. Can be improved.
+          this.isLoading$.next(false);
+          this.formErrorHandler.handleWsFormError(error, this.form);
+          this.cdr.markForCheck();
+        },
+      });
+
+    return false;
+  }
+
+  getPayload(): CloudsyncCredentialUpdate {
+    const commonValues = this.form.value;
+    return {
+      name: commonValues.name,
+      provider: commonValues.provider,
+      attributes: this.providerForm.getSubmitAttributes(),
+    };
+  }
+
+  setCredentialsForEdit(): void {
+    this.form.controls.name.clearValidators();
+    this.form.patchValue({
+      provider: this.existingCredential.provider,
+      name: this.existingCredential.name,
+    });
+
+    if (this.providerForm) {
+      this.providerForm.getFormSetter$().next(this.existingCredential.attributes);
+    }
+  }
+
+  onVerify(): void {
+    this.isLoading$.next(true);
+
+    const beforeSubmit$ = this.providerForm.beforeSubmit();
+
+    beforeSubmit$
+      .pipe(
+        switchMap(() => {
+          const { name, ...payload } = this.getPayload();
+          return this.ws.call('cloudsync.credentials.verify', [payload]);
+        }),
+        untilDestroyed(this),
+      )
+      .subscribe({
+        next: (response) => {
+          if (response.valid) {
+            this.snackbarService.success(this.translate.instant('The credentials are valid.'));
+          } else {
+            this.dialogService.error({
+              title: this.translate.instant('Error'),
+              message: response.excerpt,
+              backtrace: response.error,
+            });
+          }
+
+          this.isLoading$.next(false);
+          this.cdr.markForCheck();
+        },
+        error: (error) => {
+          this.isLoading$.next(false);
+          this.formErrorHandler.handleWsFormError(error, this.form);
+          this.cdr.markForCheck();
+        },
+      });
+  }
+
+  openAdvanced(): void {
```

Comment:
The Advanced Options button should be on the 2nd step since it takes user to the `CloudSyncFormComponent` and not the `CloudCredentialsFormCompondnt`

Reply:
Makes sense